compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -215,15 +233,14 @@ static LogicalResult setRootConfigForPadPackPipeline(func::FuncOp entryPointFn,
   // added to avoid failure for small GEMM sizes.
   auto initType = linalgOp.getDpsInitOperand(0)->get().getType();
   auto initShape = llvm::cast<ShapedType>(initType).getShape();
-  auto tileM0 = std::min((int)initShape[0], 64);
-  auto tileN0 = std::min((int)initShape[1], 64);
-  auto tileM1 = std::max((int)tileM0 / 2, 1);
-  auto tileN1 = std::max((int)tileN0 / 2, 1);
+  auto tileM0 = findLargestFactor((int)initShape[0], 64);
+  auto tileN0 = findLargestFactor((int)initShape[1], 64);
+  auto tileM1 = findLargestFactor((int)tileM0, 32);
+  auto tileN1 = findLargestFactor((int)tileN0, 32);
   auto lhsType = linalgOp.getDpsInputOperand(0)->get().getType();
   auto lhsShape = llvm::cast<ShapedType>(lhsType).getShape();
-  auto tileK0 = std::min((int)lhsShape[1], 256);
-  auto tileK1 = std::min((int)lhsShape[1] / 8, 4);
-
+  auto tileK0 = findLargestFactor((int)lhsShape[1], 256);
```

Comment:
This call might loop 256 times. I asked ChatGPT4 for a more efficient algo and it came up with an O(sqrt(max)) approach which isn't significantly more compilated. Inner loop looks something like

```
 for (int i = 2; i <= std::sqrt(num); ++i) {
        if (num % i == 0) { // If i is a factor
           auto correspondingFactor = num / i;
           // update to max of i and correspondingFactor. 
        }
 }
 ```

Reply:
not sure I follow, what does the outer loop looks like?

Reply:
as discussed offline, we are going to stick with the unoptimized version for now.